gnome-initial-setup/pages/network/gis-network-page.c

```diff
@@ -588,11 +589,47 @@ active_connections_changed (NMClient *client, GParamSpec *pspec, GisNetworkPage
   refresh_wireless_list (page);
 }
 
+static gboolean
+skip_page_for_ethernet_connection (GisNetworkPage  *self,
+                                   GNetworkMonitor *monitor)
+{
+  GisNetworkPagePrivate *priv;
+  GisAssistant *assistant;
+  gboolean is_current_page, has_connection;
+
+  priv = gis_network_page_get_instance_private (self);
+  assistant = gis_driver_get_assistant (GIS_PAGE (self)->driver);
+  is_current_page = gis_assistant_get_current_page (assistant) == GIS_PAGE (self);
+  has_connection = g_network_monitor_get_connectivity (monitor) == G_NETWORK_CONNECTIVITY_FULL;
+
+  /*
+   * Hide the Network page if a connection is available and it's not
+   * visible. This way, the user won't be even prompted about network.
+   * Otherwise, make it visible so the user can connect to a network,
+   * keeping in mind this page is only visible with a Wireless adapter.
+   */
+  if (!is_current_page)
+    gtk_widget_set_visible (GTK_WIDGET (self), !has_connection && priv->nm_device);
+
+  gis_page_set_complete (GIS_PAGE (self), has_connection);
```

Comment:
There's a really unlikely sequence of events which I think could leave this page in the wrong state:

* User advances to this page
* User ticks the “don't set up my network now” button; `sync_page_complete` is called and sets the page complete
* User plugs in a network cable – this function is called and sets the page complete (again)
* User unplugs the network cable – this function is called again and sets the page incomplete, which is incorrect

Maybe this function and `sync_page_complete` should be combined?